### PR TITLE
chore: Cart-lib migration schematics, deprecated constructors

### LINKED ISSUE #15219: Generate constructor schematics

### PR DESCRIPTION
**Pull request by plabadie**

### CODE REVIEW

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts [hunk 1]
```diff
@@ -0,0 +1,2253 @@
...
+    ],
+    addParams: [
+      {
+        className: 'ActiveCartFacade',
+        importPath: '@spartacus/cart/base/root',
+      },
+    ],
+  },
+  {
+    class: 'QuickOrderService',
```
Comment (WeizhengSap) [inline projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts:77]: there is `quick-order-form.component.migration.ts` inside `data` folder. There are some differences between this one and that (they should be merged, not just remove one of them).

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts [hunk 2]
```diff
@@ -0,0 +1,2253 @@
...
+      },
+      {
+        className: 'MultiCartFacade',
+        importPath: '@spartacus/cart/base/root',
+      },
+    ],
+  },
+  {
+    class: 'CartNotEmptyGuard',
+    importPath: '@spartacus/checkout/components',
```
Comment (WeizhengSap) [inline projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts:306]: this one is in checkout's base component

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts [hunk 3]
```diff
@@ -0,0 +1,2253 @@
...
+    addParams: [
+      {
+        className: 'ActiveCartFacade',
+        importPath: '@spartacus/cart/base/root',
+      },
+    ],
+  },
+  {
+    class: 'CheckoutAuthGuard',
+    importPath: '@spartacus/checkout/components',
```
Comment (WeizhengSap) [inline projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts:336]: this one is also in checkout base components, and `UserAccountFacade` and `GlobalMessageService` are also removed. I think, that removal is caused by checkout refactor. Not sure whether other devs will update this migration doc.

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts [hunk 4]
```diff
@@ -0,0 +1,2253 @@
...
+    ],
+    addParams: [
+      {
+        className: 'ActiveCartFacade',
+        importPath: '@spartacus/cart/base/root',
+      },
+    ],
+  },
+  {
+    class: 'CheckoutDetailsService',
```
Comment (WeizhengSap) [inline projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts:385]: This one is gone. All checkout stuff has to be checked by developers who did the checkout refactor.

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts [hunk 5]
```diff
@@ -0,0 +1,2253 @@
...
+    ],
+    addParams: [
+      {
+        className: 'ActiveCartFacade',
+        importPath: '@spartacus/cart/base/root',
+      },
+    ],
+  },
+  {
+    class: 'CheckoutGuard',
```
Comment (WeizhengSap) [inline projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts:419]: in checkout base component

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts [hunk 6]
```diff
@@ -0,0 +1,2253 @@
...
+    addParams: [
+      {
+        className: 'ActiveCartFacade',
+        importPath: '@spartacus/cart/base/root',
+      },
+    ],
+  },
+  {
+    class: 'CheckoutLoginComponent',
+    importPath: '@spartacus/checkout/components',
```
Comment (WeizhengSap) [inline projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts:462]: checkout base

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts [hunk 7]
```diff
@@ -0,0 +1,2253 @@
...
+    addParams: [
+      {
+        className: 'ActiveCartFacade',
+        importPath: '@spartacus/cart/base/root',
+      },
+    ],
+  },
+  {
+    class: 'CheckoutOrderSummaryComponent',
+    importPath: '@spartacus/checkout/components',
```
Comment (WeizhengSap) [inline projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts:492]: base

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts [hunk 8]
```diff
@@ -0,0 +1,2253 @@
...
+    ],
+    addParams: [
+      {
+        className: 'ActiveCartFacade',
+        importPath: '@spartacus/cart/base/root',
+      },
+    ],
+  },
+  {
+    class: 'CheckoutProgressMobileTopComponent',
```
Comment (WeizhengSap) [inline projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts:513]: base

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts [hunk 9]
```diff
@@ -0,0 +1,2253 @@
...
+    ],
+    addParams: [
+      {
+        className: 'ActiveCartFacade',
+        importPath: '@spartacus/cart/base/root',
+      },
+    ],
+  },
+  {
+    class: 'NotCheckoutAuthGuard',
```
Comment (WeizhengSap) [inline projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts:539]: base

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts [hunk 10]
```diff
@@ -0,0 +1,2253 @@
...
+    addParams: [
+      {
+        className: 'ActiveCartFacade',
+        importPath: '@spartacus/cart/base/root',
+      },
+    ],
+  },
+  {
+    class: 'PaymentMethodComponent',
+    importPath: '@spartacus/checkout/components',
```
Comment (WeizhengSap) [inline projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts:574]: moved to checkout's `base`, and class renamed to `CheckoutPaymentMethodComponent`;  `ChecoutFacade` removed; and`CheckoutDeliveryAddressFacade` removed and `CheckoutDeliveryFacade` added (or maybe just renamed)

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts [hunk 11]
```diff
@@ -0,0 +1,2253 @@
...
+    ],
+    addParams: [
+      {
+        className: 'ActiveCartFacade',
+        importPath: '@spartacus/cart/base/root',
+      },
+    ],
+  },
+  {
+    class: 'ReviewSubmitComponent',
```
Comment (WeizhengSap) [inline projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts:627]: this one is also moved to `base`, renamed, and need manually check its constructor.

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts [hunk 12]
```diff
@@ -0,0 +1,2253 @@
...
+    addParams: [
+      {
+        className: 'OrderFacade',
+        importPath: '@spartacus/order/root',
+      },
+    ],
+  },
+  {
+    class: 'WishListComponent',
+    importPath: '@spartacus/storefront',
```
Comment (WeizhengSap) [inline projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts:2233]: need add this "moved to wish-list entry point" to migration doc

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts [hunk 13]
```diff
@@ -0,0 +1,2253 @@
...
+      },
+      {
+        className: 'SelectiveCartFacade',
+        importPath: '@spartacus/cart/base/root',
+      },
+    ],
+  },
+  {
+    class: 'TrackingEventsComponent',
+    importPath: '@spartacus/storefront',
```
Comment (WeizhengSap) [inline projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts:2207]: move to order-lib

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts [hunk 14]
```diff
@@ -0,0 +1,2253 @@
...
+        className: 'ActiveCartService',
+        importPath: '@spartacus/core',
+      },
+    ],
+    addParams: [
+      {
+        className: 'ActiveCartFacade',
+        importPath: '@spartacus/cart/base/root',
+      },
+      {
```
Comment (WeizhengSap) [inline projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts:1679]: router is not added.

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts [hunk 15]
```diff
@@ -0,0 +1,2253 @@
...
+    ],
+    addParams: [
+      {
+        className: 'ActiveCartFacade',
+        importPath: '@spartacus/cart/base/root',
+      },
+    ],
+  },
+  {
+    class: 'AddToCartComponent',
```
Comment (WeizhengSap) [inline projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts:1324]: there are 2 `'AddToCartComponent'`, the next one is also `'AddToCartComponent'`.

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts [hunk 16]
```diff
@@ -0,0 +1,2253 @@
...
+      {
+        className: 'ActiveCartService',
+        importPath: '@spartacus/core',
+      },
+      {
+        className: 'CommonConfiguratorUtilsService',
+        importPath: '@spartacus/product-configurator/common',
+      },
+      {
+        className: 'CheckoutFacade',
```
Comment (WeizhengSap) [inline projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts:1200]: `CheckoutFacade` replaced by `CheckoutQueryFacade`

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts [hunk 17]
```diff
@@ -0,0 +1,2253 @@
...
+    ],
+    addParams: [
+      {
+        className: 'ActiveCartFacade',
+        importPath: '@spartacus/cart/base/root',
+      },
+    ],
+  },
+  {
+    class: 'PaymentTypeService',
```
Comment (WeizhengSap) [inline projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts:909]: this one also need manually checked.

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts [hunk 18]
```diff
@@ -0,0 +1,2253 @@
...
+    ],
+    addParams: [
+      {
+        className: 'ActiveCartFacade',
+        importPath: '@spartacus/cart/base/root',
+      },
+    ],
+  },
+  {
+    class: 'CheckoutService',
```
Comment (WeizhengSap) [inline projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts:875]: need manually checked

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts [hunk 19]
```diff
@@ -0,0 +1,2253 @@
...
+    ],
+    addParams: [
+      {
+        className: 'ActiveCartFacade',
+        importPath: '@spartacus/cart/base/root',
+      },
+    ],
+  },
+  {
+    class: 'CheckoutPaymentService',
```
Comment (WeizhengSap) [inline projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts:837]: need manually checked

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts [hunk 20]
```diff
@@ -0,0 +1,2253 @@
...
+    ],
+    addParams: [
+      {
+        className: 'ActiveCartFacade',
+        importPath: '@spartacus/cart/base/root',
+      },
+    ],
+  },
+  {
+    class: 'CheckoutDeliveryService',
```
Comment (WeizhengSap) [inline projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts:769]: this one is renamed, and need manually checked

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts [hunk 21]
```diff
@@ -0,0 +1,2253 @@
...
+    ],
+    addParams: [
+      {
+        className: 'ActiveCartFacade',
+        importPath: '@spartacus/cart/base/root',
+      },
+    ],
+  },
+  {
+    class: 'CheckoutCostCenterService',
```
Comment (WeizhengSap) [inline projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts:739]: need manually checked

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts [hunk 22]
```diff
@@ -0,0 +1,2253 @@
...
+    ],
+    addParams: [
+      {
+        className: 'ActiveCartFacade',
+        importPath: '@spartacus/cart/base/root',
+      },
+    ],
+  },
+  {
+    class: 'ShippingAddressComponent',
```
Comment (WeizhengSap) [inline projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts:681]: This one rename to `CheckoutDeliveryAddressComponent`. Need manually checked.

File: projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts (hunk 15 above)
Comment (plabadie) [reply projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts:1324]: They are different constructors from the same class.

**Pull request closed without merging**